linetools/lists/linelist.py

```diff
@@ -344,7 +349,195 @@ def all_transitions(self,line):
             else:
                 raise ValueError('Line {} not found in the linelist'.format(line))
 
+    def strongest_transitions(self,line, wvlims, z, n_max=3,verbose=False):
+        """For a given single line transition, this function returns
+        the n_max strongest transitions of the ion found in 
+        the linelist, within the wavelenght range wlims at a given redshift z
+
+
+        Parameters:
+        ----------
+        line: str or Quantity
+            Name of line. (e.g. 'HI 1215', 'HI', 'CIII', 'SiII', 1215.6700*u.AA)
+            [Note: when string contains spaces it only considers the first
+             part of it, so 'HI' and 'HI 1215' and 'HI 1025' are all equivalent]
+            [Note: to retrieve an unknown line use string 'unknown']
+        wvlims : tuple of Quantity
+            Wavelength range, e.g. wvlims=(1100*u.AA, 3200*u.AA) or wvlims=(1100, 3200)*u.AA
+        z : float
+            Redshift of interest
+        n_max : int 
+            Maximun number of transitions to retrieve
+
+        Returns:
+        ----------
+        None (if no transitions are found), dict (if only 1 transition found), or QTable (if > 1 transitions are found)
+
+        """    
+
+        #Check correct format
+        if isinstance(wvlims, tuple): # tuple
+            if all(isinstance(wvlim,Quantity) for wvlim in wvlims):
+                pass
+            else:
+                raise ValueError('Elements of wvlims have to be of class Quantity; correct format please')
+        elif isinstance(wvlims, Quantity): # or quantity
+            pass
+        else:
+            raise ValueError('wvlims has to be tuple or Quantity')
+        if len(wvlims) != 2:
+            raise SyntaxError('wlims has to be of size len()== 2; Please correct format')
+        if wvlims[0] >= wvlims[1]:
+            raise ValueError('Minimum limit (wlims[0]) is not smaller than maximum limit (wlims[1]); please correct')
+
+        if n_max < 1:
+            return None
+
+        data = self.all_transitions(line)
+        wv_obs = data['wrest'] * (1 + z)
+        # condition to be within wvrange
+        cond = (wv_obs >= wvlims[0]) & (wv_obs <= wvlims[1])
+        if np.sum(cond) == 0:
+            if verbose:
+                print('[strongest_transitions] Warning: no transitions found within wvlims; returning None')
+            return None
+        elif isinstance(data,dict): #Only 1 case from a dict format
+            return data
+        elif np.sum(cond) == 1: #only 1 case from a QTable format
+            name = data[cond]['name'][0]
+            return self.__getitem__(name)
+        else:
+            #remove transitions out of range
+            data = data[cond]
+            #add strength column to data
+            strength = data['wrest'] * data['f']
+            data.add_column(Column(data=strength,name='strength'))
```

Comment:
While this surely works, I think you can slice the Table, i.e. provide it a sorted list of indices.